src/pip/_internal/cli/base_command.py

```diff
@@ -257,6 +260,25 @@ def main(self, args):
 
 class RequirementCommand(Command):
 
+    @staticmethod
+    def make_requirement_preparer(directory,   # type: TempDirectory
```

Comment:
Even though the method below uses this style, I would use the following style as it's more common throughout pip:

```python
def make_requirement_preparer(
    directory,   # type: TempDirectory
    ...
```

Reply:
Also, you might as well call this `temp_directory`.

Reply:
Or should I name it `build_dir` just the way it is in `wheel.py`, `download.py` and the `RequirementPreparer` class ?

Reply:
It's always a `temp_dir` I believe (but you should check), so I would just use that for now. That would e.g. let you put `temp_dir.path` in the method instead of at each call site.

Reply:
yes its temp_directory in all of them